### PR TITLE
Add the `AutoInject` attribute injection instead of constructor injections in the TodoTemplate project (#2160)

### LINKED ISSUE #2160: Use the new `AutoInject` attribute to remove all constructor injections in the `TodoTemplate` projects
After developing the attribute and Source Generator, it is required to use this new feature in the TodoTemplate project.

### PR DESCRIPTION
**Pull request by mohabbati**
This closes #2160

### CODE REVIEW

**Comment by mohabbati** [comment]
Note that, partial classes that use `AutoInject` need to **uncheck** "Enable just my code" in "Options --> Debugging" to debug.

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
-
-    public AttachmentController(IOptionsSnapshot<AppSettings> setting,
-        UserManager<User> userManager,
-        IWebHostEnvironment webHostEnvironment)
-    {
-        _appSettings = setting.Value;
-        _userManager = userManager;
-        _webHostEnvironment = webHostEnvironment;
-    }
+    [AutoInject] private readonly AppSettings _appSettings;
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs:12]: Have you tested this? @mohabbati 
Use public properties with pascal naming

**Review by ysmoradi — CHANGES_REQUESTED** [review]

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs (hunk 1 above)
Comment (mohabbati) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs:12]: I tested it. It works well.
I'll change all fields to property.
@ysmoradi

**Comment by ysmoradi** [comment]
> Note that, partial classes that use `AutoInject` need to **uncheck** "Enable just my code" in "Options --> Debugging" to debug.

It's by design + Undo what I told you about using public properties, use private readonly _field instead

**Review by ysmoradi — APPROVED** [review]

**Review by msynk — APPROVED** [review]

**Comment by msynk** [comment]
@mohabbati 
please resolve the conflict

**Pull request merged**